syntax = "proto3";

message GatewayRequest{
    RequestType request_type =1;
    string value =2;
    int32 client_ident =3;
    string aux =4;
    enum RequestType{
        ReadStatus = 0;
        ReadSensor = 1;
        ModStatus  = 2;
        ModOnOf = 3;
        DiscoverComands = 4;
    }
}

message GadgetsResponse{
    string result=1;
    string name = 2;
    int32 sensor_ident =3;
    int32 client_ident=4;
    int32 object_status=5;
    repeated string object_comands =6;
}

message GadgetsIdent{
    string nome=1;
    string ip = 2;
    int64 port=3;
}